src/bio2bel/sources/rhea.py

```diff
@@ -0,0 +1,123 @@
+                ?side rh:contains ?participant .
+                ?participant rh:compound ?compound .
+                ?compound rh:name ?compound_name .
+                OPTIONAL {
+                    ?compound rh:reactivePart ?part_id .
+                    ?part_id rh:chebi ?chebi .
+                    ?part_id rh:name ?reactivePart .
+                } .
+                OPTIONAL {?compound rh:chebi ?chebi} .
+                OPTIONAL {?compound rh:underlyingChebi ?chebi}
+            }
+            """
+        result = g.query(q, initBindings={'side': side_uri})
+        # Get an iterable of the compounds (no need to remove duplicates since dictionary keys must be unique)
+        compounds = map(lambda x: x[0], result)
+        # Create a dictionary that will contain the nodes associated with a compound (for GenericCompounds with multiple ReactiveParts)
+        # Goal: for compounds with multiple ReactiveParts, link those reactiveParts together so they can easily be handled later
+        #   For now, we will ignore these compounds.
+        nodes_by_compound: Dict[str, List[dsl.BaseEntity]] = {c: [] for c in compounds}
+        for r in set(result):
+            compound_name, chebi_uri, reactive_part_name = r
+            # Based on the way the OPTIONAL modifier works, the SELECT query may return ?compound_name entries by themselves, without any ?chebi information
+            if not chebi_uri:
+                continue
+            # Remove the namespace from the chebi_uri to access the identifier
+            identifier = chebi_uri.replace(CH_NAMESPACE, '')
+            # Use the compound name if no reactive part is available
+            name = reactive_part_name if reactive_part_name else compound_name
+            # Build an abundance node, and append it to the node list under the correct compound
+            node = dsl.Abundance(namespace=CHEBI, name=name, identifier=identifier)
+            nodes_by_compound[compound_name].append(node)
+
+        for compound, node_list in nodes_by_compound.items():
+            node = None
+            if len(node_list) == 1:
+                node = node_list[0]
+            else:
+                # If there are multiple nodes, there must be multiple reactiveParts
+                # We have no idea how to handle this for the moment... skip it!
+                logger.debug(
+                    f'Encountered compound {compound} with multiple ReactiveParts {node_list} ...skipped.\n\
+                    PyBEL currently has no easy way to represent this type of entity, since the groupings contain meaningful information like stoichiometry and catalysis.\n \
+                    See: https://www.rhea-db.org/rhea/24960.',
+                )
+                continue
+            # Append the node to the corresponding list in participants
+            reaction_side = 'reactants' if suffix == '_L' else 'products'
+            participants[reaction_side].append(node)
```

Comment:
sorry to hold this up longer, but can you also keep the RHEA reaction identifier and reaction name (if available) and put them into the reaction node? then you can create a Reaction DSL object and add it to the graph with BELGraph.add_node_from_data(...)

Reply:
can do! for the reaction name information, should I just use the reaction equation?

Reply:
Yeah rhea seems to use that interchangeably with the Reaction name field in RDF

Reply:
Will you re-run the summary script? Now it should show some reactions that have the names in them (I think)

Reply:
I think those are the re-run results actually, but I'll give it another shot

Reply:
Updated—still no name/id

Reply:
Okay must be because I didn’t update the repr for the reaction class